README.md

```diff
@@ -21,13 +21,95 @@ This repository provides a terraform module that sets up the required infrastruc
+````
+4. Test the image built sshing your linuxkit instance
+1. Once you're ready to build a new AMI. Run the following:
+```make
+make remote-push BUILD_INSTANCE_NAME="linuxkit-build" FILE="linuxkit.yml"
+````
+You'll find the AMI ID in the command's output.
 
 #### Mechanics
 
+##### Make commands
+
+| Command | Inputs | Description |
+| ----------- | ----------- | ----------- |
+| `make remote-docker-login` | `URL`, `USERNAME`, `PASSWORD` | Build machine log in to a private Docker Registry. `URL` can be any registry url available as github, dockerhub, gcr, ecr...|
+| `make remote-sync` |  `BUILD_INSTANCE_NAME` |  Copy `./sync` local directory to build machine. |
+| `make remote-build` |  `BUILD_INSTANCE_NAME`, `FILE` |  Build the image with the given file name. Notice that `FILE` is a `./sync` relative path. |
+| `make remote-deploy` |  `BUILD_INSTANCE_NAME`, `LINUXKIT_INSTANCE_NAME`, `FILE` | Build the image with the given file name and mount it on the given linuxkit instance's root volume. Notice that the instance names are EC2 Tag Names. |
+| `make remote-push` |  `BUILD_INSTANCE_NAME`, `LINUXKIT_INSTANCE_NAME`, `FILE` |  Build the image and upload its raw file to AWS for AMI creation. Once it's done, you can find the AMI ID in output.|
+
+##### Input Variables
+
+- `linuxkit_instance_id` - The linuxkit instance under management
+- `linuxkit_bucket_name` - S3 bucket name for storing the exported linuxkit image
+- `machine_name` - Name of the build machine
+- `instance_ondemand` - Whether the build machine is ondemand or spot
+- `instance_type` - The instance type of build machine to start
+- `key_pair_name` - Key name of the Key Pair to use for the instance
+- `vpc_id` - VPC ID of the build machine to launch in
+- `subnet_id` - Subnet ID of the build machine to launch in
+- `aws_iam_role_id` - IAM Role name to be associated with the instance
+- `ebs_kms_key_arn` - ARN of the KMS key to use when linuxkit instances has encrypted volumes
+- `install` - Set of strings with versions of packages to be installed from userdata script. The block supports the following:
+  - `linuxkit_version` - Desired [linuxkit](https://github.com/linuxkit/linuxkit) version. The value can be any reference that would be accepted by the git checkout command, including branch and tag names.
+  - `babashka_version` - Desired [babashka](https://github.com/babashka/babashka) version. The value can be any reference that would be accepted by the git checkout command, including branch and tag names.
+
 ##### Usage
 
-document terraform module...
+Basic example - In your terraform code add something like this:
+
+```hcl
+resource "aws_iam_role" "build_machine" {
+  name = "linuxkit-build-role"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "ec2.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
```

Comment:
I know this is from before you started on the project, but I can't remember why it was necessary to create this role outside of the dev_cycle module. Should we move it in the module and if it is required to give specific access export it as output?

Reply:
Oh I see. Yes, good point. I'll definitely do that but I think it would be nice put this change in a new PR.

Reply:
https://github.com/vouch-opensource/linuxkit-image-development/pull/16